app.js

```diff
@@ -1,46 +1,110 @@
+        insertRows(tableName, parsedData);
+    });
+}
 
-    // Filter csv files
-    let csvFiles = files.filter(file => /(\.csv)$/.test(file));
-    if (!csvFiles.length) {
-        console.error(new Error('No CSV files found'));
-        return;
-    }
+function insertRows(tableName, parsedData) {
+    let completedQueryCount = 0;
+    parsedData.forEach((row, index, array) => {
+        let rowToInsert = [];
+        for (let col in row) {
+            rowToInsert.push(row[col]);
+        }
 
-    // Loop on each csv file in order to insert it to mysql database
-    csvFiles.forEach(file => {
-        let csvFile = fs.readFileSync(`./files/${file}`, "utf8");
-        let tableName = convertFileNameToTableName(file);
-        let connection = mysql.createConnection(DB_CONFIG);
+        connection.query(`INSERT INTO ${tableName} VALUES (${rowToInsert.map(x => {
+            x = x.replace(/"/g, '\\"');
+            return '"' + x + '"';
+        }).join()})`, error => {
+            // Close the connection once all queries are complete
+            if (++completedQueryCount === array.length) {
+                connection.end();
+            }
 
-        parse(csvFile, { columns: true }, (error, data) => {
             if (error) {
                 console.error(error);
                 return;
             }
+        });
+    });
+}
+
+
+let args = process.argv;
+
+if (args.indexOf('--files') > 0) {
+    isUserSpecified = true;
+    filesToInsert = args.slice(args.indexOf('--files') + 1);
+    let notFounFiles = [];
+    filesToInsert.forEach(file => {
+        try {
+            fs.statSync(`./${file}`);
+        } catch (error) {
+            console.error(new Error(`${file} not found!`));
+            notFounFiles.push(file);
+        }
+    });
+
+    // select files which user specified and they was found; to Insert found files and to not get error 
+    if (notFounFiles.length) {
+        filesToInsert = filesToInsert.filter(file => notFounFiles.indexOf(file) < 0);
+    }
+
+} else {
+    filesToInsert = fs.readdirSync('./files/', 'utf8');
+}
+
+// Filter csv files
+let csvFiles = filesToInsert.filter(file => /(\.csv)$/.test(file));
+if (!csvFiles.length) {
+    console.error(new Error('No CSV files found'));
+    return;
+}
+
+// Loop on each csv file in order to insert it to mysql database
 
-            const colsName = Object.keys(data[0]);
-            let createTable = `CREATE TABLE ${tableName} (`;
+csvFiles.forEach(file => {
+    let csvFile;
+    if (isUserSpecified) {
+        csvFile = fs.readFileSync(`./${file}`, "utf8");
+    } else {
+        csvFile = fs.readFileSync(`./files/${file}`, "utf8");
+    }
+
+    let tableName = convertFileNameToTableName(file);
+
+    parse(csvFile, { columns: true }, (error, data) => {
+        if (error) {
+            console.error(error);
+            return;
+        }
 
-            // take care of any column contain " or ' or space 
+        const colsName = Object.keys(data[0]);
```

Comment:
This line of code will be unnecessary if you don't use columns: true when using parse